module/zfs/dbuf.c

```diff
@@ -2341,10 +2391,15 @@ dbuf_undirty(dmu_buf_impl_t *db, dmu_tx_t *tx)
 	if (db->db_state != DB_NOFILL) {
 		dbuf_unoverride(dr);
 
-		ASSERT(db->db_buf != NULL);
-		ASSERT(dr->dt.dl.dr_data != NULL);
-		if (dr->dt.dl.dr_data != db->db_buf)
+		/*
```

Comment:
`do not need to destroy the ARC buffer`

Reply:
@behlendorf and I are looking closer to the dbuf code as there seems to be a race condition happening with current state of the work. In particular, we are seeing `ASSERT` failures and page faults in Fedora and FeeBSD 12.2 in `dbuf_sync_leaf()` -> `dbuf_write()`. We hope to figure out the issue soon and I will update this PR again.